platform_plugin_elm_credentials/api/serializers.py

```diff
@@ -0,0 +1,92 @@
+"""Pydantic models for ELMv3 data."""
+from typing import List
+from uuid import uuid4
+
+from pydantic import BaseModel, ConfigDict, Field
```

Comment:
I haven't used ConfigDict before, but this looks similar to Data Attributes: https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0049-django-app-patterns.html#data-py which is a common practice across Open edX projects. Can you take a look at it? Could that replace this ConfigDict?

Reply:
I understand why it appears convenient to go with pydantic, but you're listing just one use case for using a whole new library. When adding a new core library, we also need to think about support and maintainability in the long term. Open edX doesn't use this library: https://github.com/search?q=repo%3Aopenedx%2Fedx-platform%20pydantic&type=code, so installing it could mean breaking things in the future with an update. That's why I wouldn't take this decision lightly. Also, the community already has a best practice for serializable objects.

This time, this might not be a hard requirement. But usually, we should avoid using new libraries unless we don't have other options, mainly when installing them as a main requirement.